nms/app/packages/magmalte/app/components/context/RefreshContext.js

```diff
@@ -0,0 +1,305 @@
+        type: props.type,
+        networkId: props.networkId,
+        refreshingState: refreshingState,
+        id: props.id,
+        enqueueSnackbar: props.enqueueSnackbar,
+      }).then(response => {
+        if (response) {
+          setRefreshingState(response);
+        }
+      });
+    }, props.interval);
+
+    return () => {
+      // Update context on leave
+      clearInterval(intervalId);
+      updateContext({
+        context: ctx,
+        type: props.type,
+        refreshingState: refreshingState,
+        networkId: props.networkId,
+      });
+    };
+  }, [
+    ctx,
+    props.enqueueSnackbar,
+    props.id,
+    props.interval,
+    props.networkId,
+    props.type,
+    refreshingState,
+  ]);
+  return refreshingState;
+}
+
+export function updateContext(props: Props) {
+  if (props.refreshingState) {
+    if (props.type === 'subscriber') {
+      props.context.setState(
+        null,
+        props.refreshingState.state,
+        props.refreshingState.sessionState,
+      );
+    } else {
+      props.context.setState(
+        null,
+        props.networkId,
+        props.refreshingState.state,
+      );
+    }
+  }
+}
+
+type fetchProps = {
+  type: refreshType,
+  networkId: string,
+  //$FlowIgnore
+  refreshingState: any,
+  id?: string,
+  enqueueSnackbar?: (msg: string, cfg: {}) => ?(string | number),
+};
+async function fetchRefreshState(props: fetchProps) {
+  const {type, networkId, refreshingState, id, enqueueSnackbar} = props;
+  if (type === 'subscriber') {
+    return fetchSubscribers({
+      id: id,
+      networkId,
+      refreshingState: refreshingState,
+      type,
+    });
+  } else if (type === 'gateway') {
+    return fetchGateways({
+      id: id,
+      networkId,
+      refreshingState: refreshingState,
+      type,
+      enqueueSnackbar: enqueueSnackbar,
+    });
+  } else {
+    return fetchEnodebs({
+      id: id,
+      networkId,
+      refreshingState: refreshingState,
+      type,
+      enqueueSnackbar: enqueueSnackbar,
+    });
+  }
+}
+
+async function fetchEnodebs(props: fetchProps) {
+  const {networkId, refreshingState, id, enqueueSnackbar} = props;
+  if (id !== undefined && id !== '') {
+    let newEnb = {};
+    try {
+      newEnb = await MagmaV1API.getLteByNetworkIdEnodebsByEnodebSerial({
+        networkId: networkId,
+        enodebSerial: id,
+      });
+
+      if (Object.keys(newEnb).length) {
+        const newEnbSt = await MagmaV1API.getLteByNetworkIdEnodebsByEnodebSerialState(
+          {
+            networkId: networkId,
+            enodebSerial: id,
+          },
+        );
+        return {
+          ...refreshingState,
+          state: {
+            ...refreshingState.state,
+            enbInfo: {
+              ...refreshingState.state.enbInfo,
+              [id]: {enb_state: newEnbSt ?? {}, enb: newEnb},
+            },
+          },
+        };
+      }
+    } catch (e) {
+      enqueueSnackbar?.('failed fetching enodeb informations', {
```

Comment:
enodeb state not being present is a valid case, let's remove this snackbar error.